dask/bag/core.py

```diff
@@ -1630,8 +1648,14 @@ def from_delayed(values):
     return Bag(graph, name, len(values))
 
 
-def merge_distinct(seqs):
-    return set().union(*seqs)
+@curry
```

Comment:
It would be good to avoid `curry` here if possible.  Currying can easily generate complex situations.  I recommend either using `functools.partial` or passing keyword arguments explicitly (this would require changes to `reduction` to optionally support new `perpartition_kwargs` and `aggregate_kwargs` keyword arguments.  Short term I recommend going with partial because it's probably easier.